spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2021 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.jdbc;
+
+//[START spanner_jdbc_load_csv]
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Mutation.WriteBuilder;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.spanner.jdbc.CloudSpannerJdbcConnection;
+import java.io.FileReader;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+/** Sample showing how to load CSV file data into Spanner */
+public class LoadCsvExample {
+  public static final String EXCEL = "EXCEL";
+  public static final String POSTGRESQL_CSV = "POSTGRESQL_CSV";
+  public static final String POSTGRESQL_TEXT = "POSTGRESQL_TEXT}";
+
+  enum SpannerDataType {
+    STRING,
+    BYTES,
+    INT64,
+    FLOAT64,
+    DOUBLE,
+    LONG,
+    NUMERIC,
+    BOOL,
+    DATE,
+    TIMESTAMP
+  }
+
+  public static Boolean hasHeader = false;
+  public static Connection connection;
+  public static Map<String, SpannerDataType> tableColumns = new LinkedHashMap<>();
+
+  /** Return the data type of the column type **/
+  public static SpannerDataType parseSpannerDataType(String columnType) {
```

Comment:
It seems odd we expect users to implement this theirselves. Is there not a library method?

Reply:
Not sure what you mean by "users implement this themselves"? Could you elaborate?

Reply:
You expect the user to copy and paste 20+ lines of regex if statements just to set the data type. `parseSpannerDataType` seems like it should be provided by a client library

Reply:
I brought this up as a concern with the client team they confirmed that no client library exists that does this and this current method would seem to be the best way.

Reply:
sg